packages/block-editor/src/components/block-styles/index.js

```diff
@@ -77,7 +77,7 @@ const useGenericPreviewBlock = ( block, type ) =>
 						innerBlocks: type.example.innerBlocks,
 				  } )
 				: cloneBlock( block ),
-		[ block, type ]
+		[ type.example ? block.name : block, type ]
```

Comment:
👋  is this change related to this PR?

Reply:
It fixes the performance issue - I'll extract it into a separate PR

Reply:
https://github.com/WordPress/gutenberg/pull/22355